openedx/features/course_experience/course_updates.py

```diff
@@ -0,0 +1,112 @@
+"""
+Utilities for course updates.
+"""
+
+import hashlib
+from datetime import datetime
+
+from lms.djangoapps.courseware.courses import get_course_info_section_module
+from openedx.core.djangoapps.user_api.course_tag.api import get_course_tag, set_course_tag
+
+STATUS_VISIBLE = 'visible'
+STATUS_DELETED = 'deleted'
+
+VIEW_WELCOME_MESSAGE_KEY = 'view-welcome-message'
+
+
+def _calculate_update_hash(update):
```

Comment:
I want especially reviewers to look at my new solution for storing dismissals by hash. This is going to be something that is going to be hard to switch to a new way in the future (even hard to modify features of, like edits showing as new updates).

But it felt like the most reasonable approach for the requirements (see docstring of get_current_update_for_user).

I was leery of taking an approach that modified how studio worked (like adding a new "version" or "modified date" field to the course items) just because that felt heavier touch? But I'm open to it.